src/CrudPanel.php

```diff
@@ -237,16 +239,91 @@ public function sort($items)
      */
-    private function getRelationModel($relationString)
+    public function getRelationModel($relationString, $length = null, $model = null)
     {
-        $result = array_reduce(explode('.', $relationString), function ($obj, $method) {
+        $relationArray = explode('.', $relationString);
+
+        if (! isset($length)) {
+            $length = count($relationArray);
+        }
+
+        if (! isset($model)) {
+            $model = $this->model;
+        }
+
+        $result = array_reduce(array_splice($relationArray, 0, $length), function ($obj, $method) {
             return $obj->$method()->getRelated();
-        }, $this->model);
+        }, $model);
 
         return get_class($result);
     }
+
+    /**
+     * Get the given attribute from a model or models resulting from the specified relation string (eg: the list of streets from
+     * the many addresses of the company of a given user).
+     *
+     * @param Model $model Model (eg: user).
+     * @param string $relationString Model relation. Can be a string representing the name of a relation method in the given
+     *        Model or one from a different Model through multiple relations. A dot notation can be used to specify
+     *        multiple relations (eg: user.company.address).
+     * @param string $attribute The attribute from the relation model (eg: the street attribute from the address model).
+     *
+     * @return array An array containing a list of attributes from the resulting model.
+     */
+    public function getModelAttributeFromRelation($model, $relationString, $attribute)
+    {
+        $lastModels = $this->getRelationModelInstance($model, $relationString);
+        $attributes = [];
+        foreach ($lastModels as $model) {
+            if ($model->{$attribute}) {
+                $attributes[] = $model->{$attribute};
+            }
+        }
+
+        return $attributes;
+    }
+
+    /**
+     * Traverse the tree of relations for the given model, defined by the given relation string, and return the ending
+     * associated model instance or instances.
+     *
+     * @param Model $model The CRUD model.
+     * @param string $relationString Relation string. A dot notation can be used to chain multiple relations.
+     * @return array An array of the associated model instances defined by the relation string.
+     */
+    private function getRelationModelInstance($model, $relationString)
+    {
+        $relationArray = explode('.', $relationString);
+        $firstRelationName = array_first($relationArray);
+        $relation = $model->{$firstRelationName};
+
+        $results = [];
+        if (! empty($relation)) {
+            if ($relation instanceof Collection) {
+                $currentResults = $relation->toArray();
+            } else {
+                $currentResults[] = $relation;
+            }
+
+            array_shift($relationArray);
+
+            if (! empty($relationArray)) {
```

Comment:
Isn't this better expressed as something like `is_array()` with/without an empty?

Reply:
`is_array()` would still return true even for an empty array...